Feedback

proubatsis opened the pull request:
Addresses: #8 #16 #18

evelinec commented on README.adoc:
```diff
@@ -12,11 +12,11 @@
+:page-tags: ['microservices', 'Kubernetes', 'Docker', 'containers', 'kubectl', 'Minikube', 'MicroProfile', 'Config', 'ConfigMaps', 'Secrets']
 :page-permalink: /guides/{projectid}
-:page-related-guides: ['docker', 'istio', 'kubernetes']
+:page-related-guides: ['kubernetes', 'microprofile-config']
```
Add more related guides according to latest comment in the feedback issue.

evelinec replied:
Btw, I think the kube-intro guide projectid is "kubernetes-intro", you need this value for the related guide data.

evelinec commented on README.adoc:
```diff
@@ -64,9 +64,9 @@ include::{common-includes}/gitclone.adoc[]
+== Deploying the microservices
 
-You'll deploy two applications, `name` and `ping`. `name` displays a greeting and displays its container name. `ping` accepts a hostname as a parameter and returns `pong` if it can successfully make a request to `name`, otherwise the request fails. To build these applications, navigate to the `start` directory and run the following command.
+The two microservices you will deploy are called `name` and `ping`. The `name` microservice simply displays a brief greeting and the name of the container that it runs in. The `ping` microservice simply pings the {kube} Service that encapsulates the pod running the `name` microservice. This demonstrates how communication can be established between pods inside a cluster. To build these application ...
```
The ping microservice simply pings the {kube} Service that...
=> I think the "simply" for the `ping` microservice is not needed
=> does "Service" need to be cap'ed here?

yeekangc replied:
I had the same question regarding Service but concluded that it should since we are referring to Kubernetes Service here.

evelinec commented on README.adoc:
```diff
@@ -109,27 +109,49 @@ Navigate to `{name-api}` and use the username `bob` and the password `bobpwd` to
+include::finish/name/src/main/java/io/openliberty/guides/name/NameResource.java[tags=greeting]
+----
+
+Next, update the response to use the configured greeting:
```
perhaps to mention "in the `getContainerName` method, ie, "Next, update the response to use the configured greeting in the `getContainerName`:" to be more specific.

yeekangc commented on README.adoc:
````diff
@@ -181,11 +204,24 @@ Navigate to `{name-api}` and you will see that the greeting message has changed
 
 ```
-mvn verify -Ddockerfile.skip=true -Dcluster.ip=[hostname]
+mvn verify -Ddockerfile.skip=true -Dcluster.ip=`minikube ip`
````
Let us present `minikube ip` consistent with how we did it in the earlier section?  We used [hostname] there too?

Or, we go for the command that doesn't need cluster.ip?

proubatsis replied:
@yeekangc `minikube ip` is actually a command. If you run this as is (ie. copy and paste it), it will automatically substitute `minikube ip` for the actual ip.

yeekangc replied:
Great.  It is hard to tell what we are trying to say from the PR.

proubatsis commented:
All the comments have been resolved @yeekangc @evelinec

yeekangc commented on README.adoc:
```diff
@@ -109,27 +109,45 @@ Navigate to `{name-api}` and use the username `bob` and the password `bobpwd` to
 ----
 
-== Create ConfigMap and Secret
+The changes introduced here use MicroProfile Config and CDI to inject the value of the environment variables `USERNAME` and `PASSWORD` into the ping resource class. Then the hardcoded values are removed from those same properties.
```
If it is to explain the changes, then we probably want to keep it.

evelinec commented on README.adoc:
```diff
@@ -111,44 +111,40 @@ Similarly, navigate to `{ping-api}/name-service` and you will see `pong`.
 
 == Modifying ping microservice
 
 Update the `ping` service to make the user name and password members configurable. These values are used to authenticate with the `name` service and they are currently hardcoded to `bob` and `bobpwd`. Now, they will take the values of the environment variables `USERNAME` and `PASSWORD`.
```
I think we prefer not to use "Let us.." in the guide much. Use first person is the general preference. ie, "You will modify.."

yeekangc replied:
The key is to connect these together so users will not misinterpret as 2 different actions.

proubatsis commented:
@yeekangc @evelinec 

* Changed modifying ping and name introductions to this format: `The name service is hard coded to have Hello! as the greeting message. We will make the greeting message configurable. Replace ...`
* Reworked the snippet explanations
* Introduce no_copy role

evelinec commented:
I'm good with the changes.

yeekangc commented:
Committed some changes.  Looks good.  Please proceed to merge.  Thank you, @proubatsis and @evelinec.

yeekangc commented:
The build completed successfully and looks fine.  (There were warnings that we probably should investigate separately.)  Merging.
